includes/class-wsuwp-transfer-equivalencies.php

```diff
@@ -26,5 +36,528 @@ public static function get_instance() {
+			$results = array();
+			$results[] = '<table class="tce-courses">
+				<thead>
+					<tr>
+						<th>Transfer</th>
+						<th colspan="3">WSU Equivalent</th>
+					</tr>
+					<tr>
+						<th>Course(s)</th>
+						<th>Course(s)</th>
+						<th>Course Title</th>
+						<th>UCORE Requirement</th>
+					</tr>
+				</thead>
+				<tbody>';
+
+			foreach ( $courses as $course ) {
+				if ( '' !== $course->InternalCourses ) {
+					// Parse the value of the 'InternalCourses' key into three different chunks...
+					// Start with blanks and fill them in as we're able.
+					$wsu_course = '';
+					$wsu_title = '';
+					$wsu_ucore = '';
+
+					if ( 'NON_T NON-T Non-Transfer' === $course->InternalCourses ) {
+						$wsu_course = 'NON_T';
+						$wsu_title = 'Non-Transfer';
+					} else {
+						$wsu_info = $course->InternalCourses;
+						$title_end = strlen( $wsu_info );
+
+						// Try to find the course subject and prefix first.
+						$info_exploded = explode( ' ', $wsu_info );
+						if ( is_array( $info_exploded ) ) {
+							$wsu_course = trim( $info_exploded[0] . ' ' . $info_exploded[1] );
+						}
+
+						// Try to find if this course fulfills a UCORE requirement.
+						$ucore_start = strpos( $wsu_info, 'UCORE - ' );
+						if ( $ucore_start ) {
+							$wsu_ucore = trim( substr( $wsu_info, $ucore_start + 8, -1 ) );
+							$title_end = -( strlen( $wsu_ucore ) + 21 );
+						}
+
+						// Remove what we've already found and call what remains the title.
+						$wsu_title = trim( substr( $wsu_info, strlen( $wsu_course ), $title_end ) );
+					}
+
+					$results[] = '<tr>
+						<td>' . esc_html( $course->IncomingCourse ) . '</td>
+						<td>' . esc_html( $wsu_course ) . '</td>
+						<td>' . esc_html( $wsu_title ) . '</td>
+						<td>' . esc_html( $wsu_ucore ) . '</td>
+					</tr>';
+				}
+			}
+
+			$results[] = '</tbody></table>';
+			$results['content'] = implode( $results );
+			$results['pagination_links'] = '';
+		} else {
+			// Build institution browsing results.
+			$institution_query_args = array(
+				'orderby' => 'title',
+				'order' => 'ASC',
+				'posts_per_page' => $this->results_per_page,
+				'post_type' => $this->content_type_slug,
+
+			);
+
+			if ( isset( $_POST['index'] ) ) {
+				$institution_query_args['meta_key'] = '_tce_alpha_key';
+				$institution_query_args['meta_value'] = sanitize_text_field( $_POST['index'] );
+			}
+
+			if ( isset( $_POST['search'] ) ) {
+				$institution_query_args['s'] = sanitize_text_field( $_POST['search'] );
+			}
+
+			if ( isset( $_POST['page'] ) ) {
+				$institution_query_args['offset'] = ( $_POST['page'] - 1 ) * $this->results_per_page;
+
+				set_query_var( 'paged', sanitize_text_field( $_POST['page'] ) );
+			}
+
+			$institution_query = new WP_Query( $institution_query_args );
+
+			if ( $institution_query->have_posts() ) {
+				$institution_results = array();
+				$institution_results[] = '<ul>';
+
+				while ( $institution_query->have_posts() ) {
```

Comment:
I'm wondering if this loop and the other similar one can be abstracted to its own method somehow. They both do almost exactly the same thing, but one captures for return and one outputs. Is the biggest difference the pagination?